### PR TITLE
Adding initializer / YML file and modifying the generator

### PR DESCRIPTION
**Pull request by ennukee**
New initializer, modified generator, and skeleton YML file.

Some local testing to be done first but the code should stay (relatively) the same.

### CODE REVIEW

File: lib/generators/install/install_generator.rb [hunk 1]
```diff
@@ -1,11 +1,9 @@
 class InstallGenerator < Rails::Generators::Base
   source_root File.expand_path('../templates', __FILE__)
-  argument :objects, :type => :string, default: "b"
 
   def setup_initializer
-    # TO-DO
-    
-    # Replace the content of this method with the logic for the new
-    # install initializer as depicted on the GitHub issue #5
+    template "mse_router_info.yml", "config/mse_router_info.yml"
```
Comment (dfaulken) [inline lib/generators/install/install_generator.rb:5]: Nice, I like how readable this DSL is.

File: lib/generators/install/templates/microservices_engine.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+require 'net/http'
+config_file = '../mse_router_info.yml'
+puts("[MSE] > No router configuration YAML file found") and return if
```
Comment (dfaulken) [inline lib/generators/install/templates/microservices_engine.rb]: The trailing `if` is only for inline usage - as long as you're introducing a line break you may as well write this in standard `if` notation.

Also, `puts and return` is basically a `raise` right?

File: lib/generators/install/templates/microservices_engine.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  not File.file?(config_file)
+
+config_data = YAML.read(config_file)
+router_uri = 'http://example.com/' # TODO: Add the correct router endpoint
+res = Net::HTTP.post_form(router_uri,
+                    'ph_name' => config_data['service_name'],
```
Comment (dfaulken) [inline lib/generators/install/templates/microservices_engine.rb]: Will non-hash-rocket syntax work here? Prefer that in general.

File: lib/generators/install/templates/microservices_engine.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# Placeholders stuff below
+puts res.body # PH
+
+# Probably don't need this if the router will just send data to the endpoint
+# (not in a response, right?). Actually, that'd be way easier if I don't
+# have to handle the response.
+if res.code == 200 and res.message == "OK"
+  # We did it!
+else
+  # We suck
```
Comment (dfaulken) [inline lib/generators/install/templates/microservices_engine.rb]: Lol

**Review by dfaulken — APPROVED** [review]
Great work - could you introduce `rubocop` to the development gemset?

At some point when we're in together, let's remember to set up Travis and CodeClimate for this repo also.

**Comment by dfaulken** [comment]
I realized it might be nicer if I let you merge these incremental PRs at your leisure, rather than just doing it for you. Let me know if you disagree.

File: lib/generators/install/templates/microservices_engine.rb (hunk 3 above)
Comment (ennukee) [reply lib/generators/install/templates/microservices_engine.rb]: I want to say no but I'm not entirely sure since I haven't used this method before.

**Comment by ennukee** [comment]
I usually just PR whatever I did that day (if I finish it) then just let it sit until the next day then merge it myself if nothing major comes up. 

Unless it's something small like tweaks / file removals, then I usually just merge those instantly.

File: lib/generators/install/templates/microservices_engine.rb (hunk 2 above)
Comment (ennukee) [reply lib/generators/install/templates/microservices_engine.rb]: Doesn't that raise an Exception though (with trace stack and all)? I just want to tell the user that the config isn't found and quit out (although I don't think `return` works outside of a method now that I think about it, so raise may be the better option)

Comment (dfaulken) [reply lib/generators/install/templates/microservices_engine.rb]: It does, but it is an exceptional circumstance which means we can't initialize the Rails app, so I think it's appropriate.

Comment (ennukee) [reply lib/generators/install/templates/microservices_engine.rb]: _exception_al, ha

File: lib/generators/install/templates/microservices_engine.rb (hunk 3 above)
Comment (dfaulken) [reply lib/generators/install/templates/microservices_engine.rb]: Worth trying - it's nicer to read

**Pull request merged**